quesma/parsers/sql/lexer/dialect_sqlparse/keyword.go

```diff
@@ -0,0 +1,932 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+
+// This file contains code derived from:
+//
+// 1. sqlparse (Copyright (c) 2016, Andi Albrecht)
+//    Licensed under BSD-3-Clause License
+//    https://github.com/andialbrecht/sqlparse/blob/38c065b86ac43f76ffd319747e57096ed78bfa63/LICENSE
+
+package dialect_sqlparse
+
+import (
+	"github.com/QuesmaOrg/quesma/quesma/parsers/sql/lexer/core"
+	"maps"
+)
+
+// The following code block is based on KEYWORDS from sqlparse:
+// https://github.com/andialbrecht/sqlparse/blob/38c065b86ac43f76ffd319747e57096ed78bfa63/sqlparse/keywords.py
+
+var KEYWORDS = map[string]*core.TokenType{
```

Comment:
Is any reason to to use a pointer instead of just value as map value type?

Reply:
The reason here was that in `Token` struct the `Type` is a pointer.